### PR TITLE
Nguigs diffeo

### PR DESCRIPTION
**Pull request by nguigs**
-Backend env variable
-Condition for while loop in mean alg
-So3: -fix random uniform when point_type=matrix
          -different belong when point_type=matrix to avoid inverting: will crash if a singular matrix is passed

### CODE REVIEW

**Comment by ninamiolane** [comment]
Great, thanks!

Two comments before I look at the code:
- Next time, could you do one PR per unit of work (4 PRs in this case, each corresponding to 1 of your bullet points)?
- Any idea why the travis tests are not passing?  "All checks have failed": 
https://travis-ci.org/geomstats/geomstats/builds/626085480?utm_source=github_status&utm_medium=notification

**Comment by nguigs** [comment]
> Great, thanks!
> 
> Two comments before I look at the code:
> 
>     * Next time, could you do one PR per unit of work (4 PRs in this case, each corresponding to 1 of your bullet points)?
> 
>     * Any idea why the travis tests are not passing?  "All checks have failed":
>       https://travis-ci.org/geomstats/geomstats/builds/626085480?utm_source=github_status&utm_medium=notification
There are two issues:
- one is tensorflow 1.15.0 that needs numpy==1.16.0
- the other I think this is an issue with the installation rather than one from the PR: see https://github.com/pysal/splot/pull/48 and https://github.com/pysal/splot/issues/47.

**Comment by ninamiolane** [comment]
Try rebase on master and re-commit-ing: we have made changes to the packages versions in requirements.txt

File: geomstats/backend/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 else:
     _backend = _default_backend
-
+    os.environ['GEOMSTATS_BACKEND'] = _backend
```
Comment (ninamiolane) [inline geomstats/backend/__init__.py]: Nice! This has been changed by a recent PR though, so you need to rebase on master

File: geomstats/geometry/riemannian_metric.py [hunk 2]
```diff
@@ -328,10 +329,10 @@ def mean(self, points,
         # TODO(nina): Profile this code to study performance,
         # i.e. what to do with sq_dists_between_iterates.
         def while_loop_cond(iteration, mean, variance, sq_dist):
-            result = gs.logical_or(
+            result = ~gs.logical_or(
                 gs.isclose(variance, 0.),
                 gs.less_equal(sq_dist, epsilon * variance))
-            return result[0, 0]
+            return result[0, 0] or iteration == 0
```
Comment (ninamiolane) [inline geomstats/geometry/riemannian_metric.py]: Thank you! 🙏

File: geomstats/geometry/special_orthogonal_group.py [hunk 3]
```diff
@@ -85,9 +85,9 @@ def belongs(self, point, point_type=None):
         elif point_type == 'matrix':
             point = gs.to_ndarray(point, to_ndim=3)
             point_transpose = gs.transpose(point, axes=(0, 2, 1))
-            point_inverse = gs.linalg.inv(point)
 
-            mask = gs.isclose(point_inverse, point_transpose)
+            mask = gs.isclose(gs.matmul(point, point_transpose),
```
Comment (ninamiolane) [inline geomstats/geometry/special_orthogonal_group.py]: Nice!

**Pull request closed without merging**